chainstate/src/detail/tokens/check_utils.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::TokensError;
+use common::{
+    chain::{Block, ChainConfig, Transaction},
+    primitives::Id,
+};
+use utils::ensure;
+
+fn check_is_text_alphanumeric(str: &[u8]) -> bool {
+    match String::from_utf8(str.to_vec()) {
+        Ok(text) => text.chars().all(char::is_alphanumeric),
+        Err(_) => false,
+    }
+}
+
+pub fn is_rfc3986_valid_symbol(ch: char) -> bool {
```

Comment:
Why do we need such an invasive function to be public? This is a the kind of function you wanna hide, while you expose something that simply checks a string.

Reply:
As I mentioned in DM, I have a few tests that check URI is valid or not. For example: `nft_issuance_icon_uri_invalid`. In these tests, we try all possible chars for URI, and ensure everything fails except for alphanumeric chars and RFC3986 symbols. The problem is any RFC3986 valid symbol is not ASCII alphanumeric. And I exposed this function to use in tests. Not sure how we can hide the function and use it in tests too.